sql/sqlspec/sqlspec.go

```diff
@@ -19,14 +19,26 @@ type (
 
 	// Table holds a specification for an SQL table.
 	Table struct {
-		Name        string         `spec:",name"`
-		Qualifier   string         `spec:",qualifier"`
-		Schema      *schemahcl.Ref `spec:"schema"`
-		Columns     []*Column      `spec:"column"`
-		PrimaryKey  *PrimaryKey    `spec:"primary_key"`
-		ForeignKeys []*ForeignKey  `spec:"foreign_key"`
-		Indexes     []*Index       `spec:"index"`
-		Checks      []*Check       `spec:"check"`
+		Name        string           `spec:",name"`
+		Qualifier   string           `spec:",qualifier"`
+		Schema      *schemahcl.Ref   `spec:"schema"`
+		Embed       []*schemahcl.Ref `spec:"embed"`
+		Columns     []*Column        `spec:"column"`
+		PrimaryKey  *PrimaryKey      `spec:"primary_key"`
+		ForeignKeys []*ForeignKey    `spec:"foreign_key"`
+		Indexes     []*Index         `spec:"index"`
+		Checks      []*Check         `spec:"check"`
+		schemahcl.DefaultExtension
+	}
+
+	// Mixin holds a specification for common table attributes.
+	Mixin struct {
+		Name        string        `spec:",name"`
+		Columns     []*Column     `spec:"column"`
+		PrimaryKey  *PrimaryKey   `spec:"primary_key"`
+		ForeignKeys []*ForeignKey `spec:"foreign_key"`
+		Indexes     []*Index      `spec:"index"`
+		Checks      []*Check      `spec:"check"`
```

Comment:
Should we create a struct with the  shared fields (cols, keys, index, etc) that is embedded in Table and Mixin?